lru/lru.go

```diff
@@ -44,6 +44,11 @@ func NewWithEvictionFunc(size int, f EvictionFunc) *Cache {
 	return c
 }
 
+// SetEvictionFunc updates the eviction func
+func (c *Cache) SetEvictionFunc(f EvictionFunc) {
+	c.cache.OnEvicted = f
```

Comment:
should be thread safe and lock before setting the value?

same for `NewWithEvictionFunc`?

Reply:
> good point! added it here

was that added in a follow-up? not seeing it here

Reply:
@liggitt would you recommend both the lock/unlock in addition to the "return error if it is already set"? i can cut another PR if so

Reply:
> @liggitt would you recommend both the lock/unlock in addition to the "return error if it is already set"?

yeah... going from nil to set still requires a lock since the function can be read/used concurrently from other methods

Reply:
(and disallowing resetting it once set still sgtm)

Reply:
Follow up is here https://github.com/kubernetes/utils/pull/319